casper/src/test/scala/coop/rchain/casper/util/rholang/InterpreterUtilTest.scala

```diff
@@ -505,4 +505,47 @@ class InterpreterUtilTest
 
     tsHash should be(Some(computedTsHash))
   }
+
+  "validateBlockCheckpoint" should "pass tests involving pick" in {
```

Comment:
I ran this test against latest `dev` and it passes. Can you help me understand why the changes in ReplayRSpace are needed?

Reply:
OK, I see now that it fails randomly